src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2321,7 +2321,7 @@ bool IsMethodNeededByTypeDueToPreservedScope (MethodDefinition method)
 				// If the type is marked, we need to keep overrides of abstract members defined in assemblies
 				// that are copied.  However, if the base method is virtual, then we don't need to keep the override
```

Comment:
Not sure if it applies here but linker must keep overrides of abstract base members if the type is kept - otherwise it's invalid IL (can't have a type which is not abstract and doesn't implement all abstract members of its base type). So non-instantiated types may keep some of their virtual members due to this (although I think linker will generate a throw body for such methods).

Reply:
Well, these are static so I don't think the instantiated rule applies -- something like "substituted for a type argument" instead?

Reply:
Do we wait until we see the type as substituted for a type argument? What about MakeGenericType/MakeGenericMethod at runtime?